drivers/spi/spi_nrf5.c

```diff
@@ -0,0 +1,481 @@
+/*
+ * Copyright (c) 2015 - 2017, Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "spi.h"
+#include "nrfx_spi.h"
+
+#define SYS_LOG_DOMAIN "SPI NRF5"
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_SPI_LEVEL
+#include <logging/sys_log.h>
+
+#include "spi_context.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define SPI_NRF5_MAX_SLAVES_ON_BUS (4)
+
+typedef void (*irq_cfg_func_t)(void);
+
+/** Configuration data */
+struct spi_nrf5_cfg {
+	nrfx_spi_t spi;
+	irq_cfg_func_t irq_cfg_func;
+	struct {
+		u8_t sck;
+		u8_t mosi;
+		u8_t miso;
+		/* Pin numbers of up to 4 slave devices */
+		u8_t ss[SPI_NRF5_MAX_SLAVES_ON_BUS];
+	} psel;
+	u8_t orc;
+};
+
+/** Driver data */
+struct spi_nrf5_data {
+	struct spi_context ctx;
+	u16_t transfer_len;
+	bool busy;
+};
+
+#define DEV_CFG(spi_cfg) ((const struct spi_nrf5_cfg * const) \
+					(spi_cfg->dev)->config->config_info)
+#define DEV_DATA(spi_cfg) ((struct spi_nrf5_data * const) \
+					(spi_cfg->dev)->driver_data)
+
+static inline nrfx_spi_frequency_t spi_freq_to_nrfx_freq(u32_t frequency)
+{
+	switch (frequency) {
```

Comment:
Would it not be better to use closed matching frequency instead of expecting the caller to give the exact frequency.